src/microbe_stage/Spawner.cs

```diff
@@ -1,50 +1,113 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 using Godot;
 
 /// <summary>
 ///   Spawner that can be added to a SpawnSystem to be used for spawning things
 /// </summary>
 public abstract class Spawner
 {
+    private float[]? binomialValuesCache;
+
+    public virtual int BinomialN => 0;
+    public virtual float BinomialP => 0;
```

Comment:
Ok, there is the explanation for the factorial args. However, would we not want to swap that to an actual utils class such as the math one?